vllm/outputs.py

```diff
@@ -75,6 +75,7 @@ def __init__(
         finished: bool,
         metrics: Optional[RequestMetrics] = None,
         lora_request: Optional[LoRARequest] = None,
+        embed = None
```

Comment:
IMO it's a better idea to create a separate class for output of embedding models. Many attributes of `RequestOutput` are tied to language models but not relevant to embedding models. We can discuss this in the later stages of this PR

Reply:
Agree @ywang96 

I think we will want to refactor `RequestOutput` and `SamplingParams` to be abstract with separate versions for `Completion` vs `Embedding`

It will be tricky to do this without breaking changes to the UX